data-processing-lib/spark/src/data_processing_spark/runtime/spark/transform_orchestrator.py

```diff
@@ -0,0 +1,135 @@
+        return 1
+    # initialize Spark
+    conf = (SparkConf().setAppName(runtime_config.get_name())
+            .set('spark.driver.host', '127.0.0.1'))
+    sc = SparkContext(conf=conf)
+    execution_params = sc.broadcast(runtime_config.get_input_params())
+    transform_class = sc.broadcast(runtime_config.get_transform_class())
+    daf = sc.broadcast(data_access_factory)
+
+    def process_partition(iterator):
+        """
+        process partitions
+        :param iterator: iterator of records
+        :return:
+        """
+        # create data access
+        d_access = daf.value.create_data_access()
+        # local statistics dictionary
+        statistics = {}
+        # create file processor
+        file_processor = SparkTransformFileProcessor(d_access=d_access, transform_params=execution_params.value,
+                                                     statistics=statistics, transform_class=transform_class.value)
+        first = True
+        for f in iterator:
+            # for every file
+            if first:
+                logger.debug(f"partition {f}")
+                # create transform with partition number
+                file_processor.create_transform(partition=int(f[1]))
+                first = False
+            # process file
+            file_processor.process_file(f_name=f[0])
+        # flush
+        file_processor.flush()
+        # return partition's statistics
+        return list(statistics.items())
+
+    try:
+        # Get files to process
+        files, profile, retries = data_access.get_files_to_process()
+        if len(files) == 0:
+            logger.error("No input files to process - exiting")
+            return 0
+        logger.info(f"Number of files is {len(files)}, source profile {profile}")
+        # process data
+        logger.debug("Begin processing files")
+        # process files split by partitions
+        logger.debug(f"parallelization {execution_config.parallelization}")
+        if execution_config.parallelization > 0:
+            stats_rdd = (sc.parallelize(files, execution_config.parallelization)
+                         .zipWithIndex().mapPartitions(process_partition))
+        else:
+            stats_rdd = sc.parallelize(files).zipWithIndex().mapPartitions(process_partition)
+        # build overall statistics
+        stats = dict(stats_rdd.reduceByKey(lambda a, b: a+b).collect())
+        return_code = 0
+        status = "success"
+    except Exception as e:
+        # process execution exception
+        logger.error(f"Exception during execution {e}: {traceback.print_exc()}")
+        return_code = 1
+        status = "failure"
+        stats = {}
+    try:
+        # build and save metadata
+        logger.debug("Building job metadata")
+        input_params = runtime_config.get_transform_metadata() | execution_config.get_input_params()
+        metadata = {
+            "pipeline": execution_config.pipeline_id,
```

Comment:
is the "parallelization" being captured as metadata somewhere?

Reply:
Yes. input parameters